### PR TITLE
add functionality to allow for multiple TEC attendance picture submission

### PR DESCRIPTION
**Pull request by alyssayzhang**
### Summary 

- [x] implemented Add button so a member can upload multiple TEC attendance for an event
- [x] add a preview button for pending TEC that member can view in a modal
- [x] add error handling if use does not upload all files 

Screenshot 2024-03-03 at 7 18 17 PM
Screenshot 2024-03-03 at 7 18 55 PM
Screenshot 2024-03-03 at 7 19 10 PM
Screenshot 2024-03-03 at 7 19 51 PM

### Notion/Figma Link 
https://www.notion.so/Idol-FA23-4899da48384747ba8ef8366f034e5ebf?p=ec88c8354f054dbb868a3e4f91f37c93&pm=s

### Test Plan 
I tested a multiple image submission for a TEC event and previewed that they were the correct images in the pending section. Also, I tested that the error message would appear for an unsuccessful image(s) upload.

### CODE REVIEW

**Review by kevinmram — COMMENTED** [review]
The progress towards enhancing TEC attendance feature by introducing the capability for multiple picture submissions is lookin' good; including an "Add" button for uploading several images is a good approach! Maybe, adding error-handling for image uploads and integrating a preview feature for users to review their submissions before finalizing could be implemented!

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx [hunk 1]
```diff
@@ -70,6 +80,24 @@ const TeamEventCreditForm: React.FC = () => {
...
+          reason: '',
+          uuid: ''
+        };
+
+        const createdAttendance = await requestTeamEventCredit(
+          newTeamEventAttendance,
+          images[index]
+        );
+
       const newTeamEventAttendance: TeamEventAttendance = {
```
Comment (andrew032011) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx:101]: Do we need both `newTeamEventAttendance` and `newTeamEventCreditAttendance` here?

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx [hunk 2]
```diff
@@ -70,6 +80,24 @@ const TeamEventCreditForm: React.FC = () => {
...
       });
     } else {
+
+      images.map(async (image, index) => {
+        const newTeamEventCreditAttendance: TeamEventAttendance = {
+          member: userInfo,
+          hoursAttended: teamEvent.hasHours ? Number(hours) : undefined,
+          image: `eventProofs/${getNetIDFromEmail(userInfo.email)}/${new Date().toISOString()}`,
+          eventUuid: teamEvent.uuid,
+          pending: true,
```
Comment (andrew032011) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx:90]: Type checker should catch this: we killed off the `pending` field already in https://github.com/cornell-dti/idol/pull/575.

**Review by andrew032011 — CHANGES_REQUESTED** [review]

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx (hunk 1 above)
Comment (alyssayzhang) [reply frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx:101]: @andrew032011 the check that did not pass said `cannot redeclare block-scoped variable 'newteameventattendance'`.

Comment (andrew032011) [reply frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx:101]: Right, but those two variables look almost the exact same, do we need both?

Comment (alyssayzhang) [reply frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx]: Ah, I see what you meant! I removed one of them as I was cleaning up code

**Review by andrew032011 — CHANGES_REQUESTED** [review]
Found a bug, here's how to reproduce:
1. Increase the number of images upload slots to 2
2. Upload image for first
3. Upload a different image for second
4. Submit
5. observe that both images are the same

It has to be done in that order to reproduce it. If you switch around 1 and 2, you won't be able to reproduce it.

Attached a video too to show you what I mean: 

https://github.com/cornell-dti/idol/assets/59291082/bf36ab96-e374-4318-966f-0d4df3d6dd97

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.module.css [hunk 3]
```diff
@@ -57,3 +57,11 @@
   display: flex;
   justify-content: flex-end;
 }
+
+.modalContent {
+  max-height: 60vh !important;
+}
+
+.margin_before_button {
```
Comment (andrew032011) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.module.css:65]: nit: camel case

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx [hunk 4]
```diff
@@ -202,16 +216,32 @@ const TeamEventCreditForm: React.FC = () => {
...
-          <input
-            id="newImage"
-            type="file"
-            accept="image/png, image/jpeg"
-            defaultValue=""
-            value={image ? undefined : ''}
-            onChange={handleNewImage}
-          />
+          {images.map((image, i) => (
+            <div className="input_container" style={{ marginBottom: '10px' }} key={i}>
```
Comment (andrew032011) [inline frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.tsx:220]: nit: inline the styling here. also, what's className "input_container"? I don't see it defined in a CSS file

**Review by andrew032011 — COMMENTED** [review]
Can you also QA test this a bit on the deploy preview just to make sure it's working now?

File: frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.module.css (hunk 3 above)
Comment (alyssayzhang) [reply frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.module.css:65]: @andrew032011 the rest of the css file is snake case

Comment (andrew032011) [reply frontend/src/components/Forms/TeamEventCreditsForm/TeamEventCreditsForm.module.css:65]: oh ok! let's leave it

**Review by andrew032011 — APPROVED** [review]

**Pull request merged**